ADA/databricks_fundamentals.qmd

```diff
@@ -0,0 +1,198 @@
+
+This means you can perform very heavy analyses that your laptop wouldn't be able to cope with. The DataBricks platform takes the place of the keyboard/mouse and screen, taking input and providing output back to the user.
+
+The storage and computation are separated into different components rather than being within the same 'machine'. Processing (processor and memory) is handled by a 'compute' resource, and storage (hard drive) is centralised in the 'unity catalog'.
+
+![](/images/ada-cloud-computing.jpg){width="515"}
+
+### Benefits of cloud compute
+
+-   **Scalable** - if you need more computing power you can increase your computing power and only pay for what you use rather than having to build an expensive new machine
+-   **Centralised** - All data, scripts, and processes are available in a single place and access for any other user can be controlled as required.
+-   **Data Governance** - The Department is able to 'see' all of it's data and organisational knowledge. This enables to ensure it is access controlled and align with GDPR and data protection legislation and guidelines.
+-   **Auditing and version control** - The Platform itself generates a lot of metadata which enables it to keep versioned history of it's data, outputs, etc.
+
+------------------------------------------------------------------------
+
+# Key concepts
+
+------------------------------------------------------------------------
+
+## Storage
+
+------------------------------------------------------------------------
+
+There are a few different ways of storing files and data on DataBricks. Your data, and modelling areas will reside in the 'unity catalog', whereas your scripts and code will live on your 'workspace'.
+
+### Unity catalog
+
+The majority of data and files on DataBricks should be stored in the 'unity catalog'. This is similar in concept to a traditional database server, however the unity catalog also contains file storage in the form of volumes.
+
+The unity catalog can be accessed through the 'Catalog' option in the DataBricks sidebar.
+
+![](/images/ada-unity-catalog-sidebar.png)
+
+#### Catalogs not databases
+
+The unity catalog contains numerous catalogs, which are similar in concept to a SQL database.
+
+#### Schemas, tables and views
+
+Like a SQL database a catalog has schemas, tables, and views which store data in a structured (usually tabular) format.
+
+This is where you would store your core datasets and pick up data to analyse from.
+
+#### Volumes
+
+Unlike a SQL database the unity catalog also contains volumes, which are file storage similar to a shared drive. They can be used for storing any type of file.
+
+Volumes are stored under a schema within a catalog. Files in here can be accessed and manipulated through code. You can also upload files to a volume through the user interface.
+
+#### Structure of the unity catalog
+
+There is one unity catalog which can contain any number of catalogs.
```

Comment:
First yes... kind of. Technically there is one unity catalog for each instance of DataBricks (dev, test, prod, pre-prod) but most analysts will only be working on Prod because it's the only environment we're allowed real data in.

Second is more complicated to answer because there are a lot of catalogs for data processing, etc. I think too much detail here could end up just bamboozling people if we start going into the 'medallion architecture' in too much detail.

Reply:
I've amended this sentence a bit more for (hopefully) clarity, but I'm hesitant to start trying to go into the specific detail of how the Department is using it here. It is useful information (not that I know enough detail about it to accurately explain - we'll want the ADA team to contribute for that I think), but could be overwhelming.